src/momento/internal/codegen.py

```diff
@@ -0,0 +1,111 @@
+"""Generate synchronous modules from asynchronous modules.
+
+Because many synchronous modules are identical to the asynchronous ones, modulo the async
+verbiage, we opt for code generation to produce parts of the synchronous code. Importantly,
+by doing this we reduce errors and increase productivity.
```

Comment:
this is only used as a build step, correct?  If so, could we put it in a different root dir and exclude it from being bundled with the distributed package?

Reply:
I'll set up a draft PR with the changes shortly. In the mean time I recommend taking a look at `test_momento.py` or `test_momento_async.py`

Reply:
very slight pref for putting it in a separate dir called build-src or something but it is not a blocker.  could also do in a separate pr later if we decide to.